core/src/main/java/hudson/XmlFile.java

```diff
@@ -205,9 +206,20 @@ public String toString() {
      * Opens a {@link Reader} that loads XML.
      * This method uses {@link #sniffEncoding() the right encoding},
      * not just the system default encoding.
+     * @throws IOException Encoding issues
+     * @return Reader for the file. should be close externally once read.
      */
     public Reader readRaw() throws IOException {
-        return new InputStreamReader(new FileInputStream(file),sniffEncoding());
```

Comment:
seems to be the cause is using a string as a charset - just use a `Charset` so no streams are created unless you have a valid charset?

```java
String encoding = sniffEncoding();
  Charset charset = null;
try { 
  charset = Charset.forName(encoding);
}
catch (UnsupportedCharsetException | IllegalArgumentException | IllegalCharsetNameException ex) {
  throw new IOException("Failed to open file for reading", ex);
}
  FileInputStream fileInputStream = new FileInputStream(file);
  return new InputStreamReader(fileInputStream, charset);
```